src/qibolab/parameters.py

```diff
@@ -50,6 +54,38 @@ def fill(self, options: ExecutionParameters):
         return options
 
 
+class TwoQubitContainer(dict[QubitPairId, TwoQubitNatives]):
```

Comment:
This class looks like a lot of code overhead, that is also quite obscure for someone without experience with pydantic, just to handle the pairs of symmetric two-qubit gates, but at this stage I trust that there was no simpler solution.

Reply:
> This class looks like a lot of code overhead, that is also quite obscure for someone without experience with pydantic, just to handle the pairs of symmetric two-qubit gates, but at this stage I trust that there was no simpler solution.

Yes, I also don't like it. And I even tried the alternative, as written above https://github.com/qiboteam/qibolab/pull/983#issuecomment-2291091875

I hoped Pydantic had a better interface for subclassing builtins, but that's the easiest. Already importing `pydantic_core` is a bit of failure from the point of view of maintainability (it's truly their `qibo_core`, btw Pydantic people are those maintaining PyO3, the framework we're also using for `qibo_core`).

But that's truly the easiest way I found :disappointed: